src/masternodes/rpc_vault.cpp

```diff
@@ -1146,6 +1146,104 @@ UniValue listauctionhistory(const JSONRPCRequest& request) {
+                },
+               RPCExamples{
+                       HelpExampleCli("estimateloan", R"(5474b2e9bfa96446e5ef3c9594634e1aa22d3a0722cb79084d61253acbdf87bf '{"TSLA":0.5, "FB": 0.4, "GOOGL":0.1}' 150)") +
+                       HelpExampleRpc("estimateloan", R"("5474b2e9bfa96446e5ef3c9594634e1aa22d3a0722cb79084d61253acbdf87bf", {"TSLA":0.5, "FB": 0.4, "GOOGL":0.1}, 150)")
+               },
+    }.Check(request);
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VOBJ, UniValue::VNUM}, false);
+
+    CVaultId vaultId = ParseHashV(request.params[0], "vaultId");
+
+    LOCK(cs_main);
+    auto height = ::ChainActive().Height();
+
+    auto vault = pcustomcsview->GetVault(vaultId);
+    if (!vault) {
+        throw JSONRPCError(RPC_DATABASE_ERROR, strprintf("Vault <%s> not found.", vaultId.GetHex()));
+    }
+
+    auto vaultState = GetVaultState(vaultId, *vault);
+    if (vaultState == VaultState::InLiquidation) {
+        throw JSONRPCError(RPC_MISC_ERROR, strprintf("Vault <%s> is in liquidation.", vaultId.GetHex()));
+    }
+
+    auto scheme = pcustomcsview->GetLoanScheme(vault->schemeId);
+    uint32_t ratio = scheme->ratio;
+    if (request.params.size() > 2) {
+        ratio = (size_t) request.params[2].get_int64();
+    }
+
+    auto collaterals = pcustomcsview->GetVaultCollaterals(vaultId);
+    if (!collaterals) {
+        throw JSONRPCError(RPC_MISC_ERROR, "Cannot estimate loan without collaterals.");
+    }
+
+    auto blockTime = ::ChainActive().Tip()->GetBlockTime();
+    auto rate = pcustomcsview->GetLoanCollaterals(vaultId, *collaterals, height + 1, blockTime, false, true);
+    if (!rate.ok) {
+        throw JSONRPCError(RPC_MISC_ERROR, rate.msg);
+    }
+
+    auto collValue = ValueFromUint(rate.val->totalCollaterals);
+    UniValue ret(UniValue::VARR);
+    CBalances loanBalances;
+    CAmount totalSplit{0};
+    if (request.params.size() > 1 && request.params[1].isObject()) {
+        for (const auto& tokenId : request.params[1].getKeys()) {
+            CAmount split = AmountFromValue(request.params[1][tokenId]);
+
+            auto token = pcustomcsview->GetToken(tokenId);
+            if (!token) {
+                throw JSONRPCError(RPC_DATABASE_ERROR, strprintf("Token %d does not exist!", tokenId));
+            }
+
+            auto loanToken = pcustomcsview->GetLoanTokenByID(token->first);
+            if (!loanToken) {
+                throw JSONRPCError(RPC_DATABASE_ERROR, strprintf("(%s) is not a loan token!", tokenId));
+            }
+
+            auto priceFeed = pcustomcsview->GetFixedIntervalPrice(loanToken->fixedIntervalPriceId);
+            if (!priceFeed.ok) {
+                throw JSONRPCError(RPC_DATABASE_ERROR, priceFeed.msg);
+            }
+
+            auto price = priceFeed.val->priceRecord[0];
+            if (!priceFeed.val->isLive(pcustomcsview->GetPriceDeviation())) {
+                throw JSONRPCError(RPC_MISC_ERROR, strprintf("No live fixed price for %s", tokenId));
```

Comment:
loanToken->symbol is better fit? At least tokenId.v is int not string.

Reply:
Here tokenId is string already